### PR TITLE
Bump linters

### PR DESCRIPTION
**Pull request by pquentin**

### CODE REVIEW

File: src/urllib3/util/retry.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def __init__(
         self.history = history or ()
         self.respect_retry_after_header = respect_retry_after_header
         self.remove_headers_on_redirect = frozenset(
-            [h.lower() for h in remove_headers_on_redirect]
+            h.lower() for h in remove_headers_on_redirect
```
Comment (pquentin) [inline src/urllib3/util/retry.py:238]: This is technically slower but unlikely to matter in practice

**Review by hramezani — APPROVED** [review]

File: src/urllib3/util/retry.py (hunk 1 above)
Comment (sethmlarson) [reply src/urllib3/util/retry.py:238]: Which tool does this change come from? We should file an issue that this construction shouldn't be modified.

Comment (sethmlarson) [reply src/urllib3/util/retry.py:238]: My eyes failed me, don't know how I read this as a `str.join()` case. This is likely faster.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**